backend/src/main/java/com/google/sps/notifs/EmailNotifier.java

```diff
@@ -0,0 +1,178 @@
+public class EmailNotifier extends Notifier {
+
+    private static final String APPLICATION_NAME = "Ad-libs";
+    private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+    private static final String TOKENS_DIRECTORY_PATH = "tokens";
+    /**
+     * Global instance of the scopes required by this quickstart. If modifying these scopes, delete
+     * your previously saved tokens/ folder.
+     */
+    private static final List<String> SCOPES =
+            Collections.singletonList(GmailScopes.MAIL_GOOGLE_COM);
+
+    private static final String CREDENTIALS_FILE_PATH = "credentials.json";
+
+    /** Variable representing the email to be notified */
+    protected String toEmail;
+
+    /**
+     * @param name Name of the recipient of the user
+     * @param to Email that this email is going to sent to.
+     */
+    public EmailNotifier(String name, String to) {
+        super(name);
+        this.toEmail = to;
+    }
+
+    /**
+     * Creates an authorized Credential object.
+     *
+     * @param HTTP_TRANSPORT The network HTTP Transport.
+     * @return An authorized Credential object.
+     * @throws IOException If the credentials.json file cannot be found.
+     */
+    private static Credential getCredentials(final NetHttpTransport HTTP_TRANSPORT)
+            throws IOException {
+        // Load client secrets.
+        InputStream in = new FileInputStream(CREDENTIALS_FILE_PATH);
+        GoogleClientSecrets clientSecrets =
+                GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+
+        // Build flow and trigger user authorization request.
+        GoogleAuthorizationCodeFlow flow =
+                new GoogleAuthorizationCodeFlow.Builder(
+                                HTTP_TRANSPORT, JSON_FACTORY, clientSecrets, SCOPES)
+                        .setDataStoreFactory(
+                                new FileDataStoreFactory(new java.io.File(TOKENS_DIRECTORY_PATH)))
+                        .setAccessType("online")
+                        .build();
+        LocalServerReceiver receiver = new LocalServerReceiver.Builder().setPort(8000).build();
+        return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
+    }
+
+    /**
+     * Create a MimeMessage using the parameters provided.
+     *
+     * @param to email address of the receiver
+     * @param subject subject of the email
+     * @param bodyText body text of the email
+     * @return the MimeMessage to be used to send email
+     * @throws MessagingException
+     */
+    public static MimeMessage createEmail(String to, String subject, String bodyText)
+            throws MessagingException {
+
+        Properties props = new Properties();
+        Session session = Session.getDefaultInstance(props, null);
+
+        MimeMessage email = new MimeMessage(session);
+
+        email.setFrom(new InternetAddress(to));
```

Comment:
Hmm should setFrom() have the "to" email as an argument?

Reply:
Oh I see, for testing, you set the from and to emails to your email. Maybe make sure to keep "to" and "from" separate and just set each as your email to test.

Reply:
Currently, only authenticated accounts can send emails and currently I'm the only authenticated sender so whichever parameter that is passed will result in me sending the email but ideally I want to use this parameter to send a dummy email

Reply:
I see. Still, I think it's important to distinguish between the from and to emails even if they are the same thing right now.